util/src/network/host.rs

```diff
@@ -321,57 +288,98 @@ struct ProtocolTimer {
 /// Root IO handler. Manages protocol handlers, IO timers and network connections.
 pub struct Host<Message> where Message: Send + Sync + Clone {
 	pub info: RwLock<HostInfo>,
-	udp_socket: Mutex<UdpSocket>,
 	tcp_listener: Mutex<TcpListener>,
-	connections: Arc<RwLock<Slab<SharedConnectionEntry>>>,
```

Comment:
`connections` was split into `handshakes` and `sessions` to simplify code in many places